common/lib/api-client/models.test.js

```diff
@@ -136,36 +153,60 @@ const client = proxyquire('./', {
   },
 })()
 
+const getResponse = ({ modelName, model, testCase } = {}) => {
+  const { statusCode = 200 } = testCase
+  const apiPath = get(model, 'options.collectionPath') || pluralize(modelName)
+  const fixture = require(`${fixturesPath}/${modelName}.${testCase.method}.json`)
+
+  const nockedResponse =
+    fixture.response !== undefined ? fixture.response : fixture
+
+  nock(mockConfig.API.BASE_URL)
+    .intercept(`/${apiPath}${testCase.mockPath || ''}`, testCase.httpMock)
+    .reply(statusCode, nockedResponse)
+
+  return client[testCase.method](modelName, testCase.args)
+}
+
 describe('API client models', function() {
   forEach(models, (model, modelName) => {
     describe(`${startCase(modelName)} model`, function() {
       forEach(testCases[modelName], testCase => {
-        describe(`#${testCase.method}()`, function() {
-          beforeEach(function() {
-            const apiPath =
-              get(model, 'options.collectionPath') || pluralize(modelName)
-            const fixture = require(`${fixturesPath}/${modelName}.${testCase.method}.json`)
+        describe(`#${testCase.method}() with 200 response`, function() {
+          if (testCase.statusCode !== 200) {
+            return
+          }
+
+          let response
 
-            nock(mockConfig.API.BASE_URL)
-              .intercept(
-                `/${apiPath}${testCase.mockPath || ''}`,
-                testCase.httpMock
-              )
-              .reply(200, fixture)
+          beforeEach(async function() {
+            response = await getResponse({ modelName, model, testCase })
           })
 
           it('should contain correct attributes', async function() {
-            const response = await client[testCase.method](
-              modelName,
-              testCase.args
-            )
-
             flatten([response.data]).forEach(item =>
               checkProperties(item, model.attributes)
             )
           })
         })
       })
+      forEach(testCases[modelName], testCase => {
+        describe(`#${testCase.method}() with 204 response`, function() {
```

Comment:
Maybe combine these tests and include the response code in the test description. It feels like this approach could become a pain if we add further response codes. We'd have to add another loop and describe going over all tests cases and return for tests that don't match the response code for that describe block.

You could then include the logic in the `it` block that for `204` it would expect no output. All other response codes then expect an output.

That also feels better because in reality we should update the POST tests to have a response code of `201` instead of `200`.